airflow-core/src/airflow/api_fastapi/core_api/security.py

```diff
@@ -255,21 +255,28 @@ def inner(
         request: BulkBody[PoolBody],
         user: GetUserDep,
     ) -> None:
+        existing_pool_names = [
+            cast("str", entity) if action.action == BulkAction.DELETE else cast("PoolBody", entity).pool
+            for action in request.actions
+            for entity in action.entities
+            if action.action != BulkAction.CREATE
+        ]
+        teams = Pool.get_bulk_team_name(existing_pool_names)
+
         requests: list[IsAuthorizedPoolRequest] = []
         for action in request.actions:
-            requests.extend(
-                [
-                    {
-                        "method": MAP_BULK_ACTION_TO_AUTH_METHOD[action.action],
-                        "details": PoolDetails(
-                            name=cast("str", pool)
-                            if action.action == BulkAction.DELETE
-                            else cast("PoolBody", pool).pool
-                        ),
-                    }
-                    for pool in action.entities
-                ]
-            )
+            for pool in action.entities:
+                pool_name = (
+                    cast("str", pool) if action.action == BulkAction.DELETE else cast("PoolBody", pool).pool
+                )
+                req: IsAuthorizedPoolRequest = {
+                    "method": MAP_BULK_ACTION_TO_AUTH_METHOD[action.action],
+                    "details": PoolDetails(
+                        name=pool_name,
+                        team_name=teams.get(pool_name),
+                    ),
+                }
+                requests.append(req)
```

Comment:
I know this logic is probably quite mechanical and obvious if you're used to this area of the codebase. But I find it a bit hard and obscure to review/read. I wonder if it's worth adding more comments to this logic going forward?